src/basic/user-util.c

```diff
@@ -423,12 +422,15 @@ int in_gid(gid_t gid) {
         if (!gid_is_valid(gid))
                 return -EINVAL;
 
-        ngroups_max = sysconf(_SC_NGROUPS_MAX);
-        assert(ngroups_max > 0);
+        ngroups = getgroups(0, NULL);
+        if (ngroups < 0)
+                return -errno;
+        if (ngroups == 0)
+                return 0;
 
-        gids = newa(gid_t, ngroups_max);
+        gids = newa(gid_t, ngroups);
```

Comment:
I think using `new()` here is safer.

Reply:
Yep, we `new` is better here. But we can consider this an "independent fix" — after all, this patch improves the situation already. So let's merge this one, and change to new in a later patch.